nomenclature for Software upgrades (#31)
onflow/Flow-Working-Groups: core_protocol_working_group/meetings/2025-02-27.md

_The new mechanism will allow a view-based upgrade for all node types. The current method of HCU will be deprecated in favor of this new method._

#### Proposed Naming Convention for Classifying Upgrades

With the introduction of this new mechanism, we now have broadly four categories to upgrade nodes on Flow as listed below.
There are three important stakeholders who should be informed about these upgrades: the flow foundation team, node operators and the community at large.

Hence, _a consistent and effective naming convention for these four type of upgrade is needed_. 

**

> 1. **Optional Upgrade**

Reviewer: @AlexHentschel so we want to retain `Optional Upgrade` ?

          and [`Node Operations` topic](https://forum.flow.com/c/node-operations/36) on Forums
        - for Community: post message "network will be down for 30 mins"

#### Discussion

* The key difference between 1 and 2 is:
   * For 2, we want to set a hard deadline, by which nodes that haven't upgraded will stop participating
   * For upgrades type 1, we generally want to encourage node operators to upgrade. Hence, we shouldn't be highlighting the fact that the update
     is non-mandatory in the headline. The message we want to convey: node operators should be updating and installing this upgrade as part of their regular maintenance cycles; 

>      but they don't have to drop other work that urgent for them, or work outside of normal business hours. 

Reviewer: ```suggestion
     but they don't have to drop other more urgent work or work outside of normal business hours. 
```
